core/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java

```diff
@@ -34,17 +36,26 @@ public interface PhysicsRollingStock {
 
     /** Get the effort the train can apply at a given speed, in newtons */
     static double getMaxEffort(double speed, TractiveEffortPoint[] tractiveEffortCurve) {
-        double previousEffort = 0.0;
-        double previousSpeed = 0.0;
-        for (var dataPoint : tractiveEffortCurve) {
-            if (previousSpeed <= Math.abs(speed) && Math.abs(speed) < dataPoint.speed()) {
-                var coeff = (previousEffort - dataPoint.maxEffort()) / (previousSpeed - dataPoint.speed());
-                return previousEffort + coeff * (Math.abs(speed) - previousSpeed);
-            }
-            previousEffort = dataPoint.maxEffort();
-            previousSpeed = dataPoint.speed();
+        //sort tractiveEffortCurve by speed
+        Arrays.sort(tractiveEffortCurve, Comparator.comparing(tractiveEffortPoint -> speed));
+        double[] tractiveEffortCurveSpeeds =
+                Arrays.stream(tractiveEffortCurve).mapToDouble(TractiveEffortPoint::speed).toArray();
```

Comment:
This line breaks performance and makes the patch less efficient than before I think.